src/Media.Plugin.iOS/PhotoLibraryAccess.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreImage;
+using Foundation;
+using Photos;
+
+namespace Plugin.Media
+{
+    public class PhotoLibraryAccess
+    {
+        public static async Task<NSDictionary> GetPhotoLibraryMetadata(NSUrl url)
+        {
+            NSDictionary meta = null;
+            var completed = false;
+            var options = new PHContentEditingInputRequestOptions
+            {
+                NetworkAccessAllowed = true,
+            };
+            NSUrl[] urls = { url };
+            var image = PHAsset.FetchAssets(urls, new PHFetchOptions()).firstObject as PHAsset;
+            await Task.Run(() =>
```

Comment:
okay. after digging for a while I have found even better way to fetch the images on iOS. just pushed it. it's now synchronous and does not require that tasks awaiting/canceling/timeouting magic. I hope you like it better as I do. It is smaller and much cleaner.